doc/source/index.rst

```diff
@@ -21,6 +21,16 @@ API Reference
    io
    util
 
+User Documentation
+------------------
+
+The user documentation contains high-level information for users of scikit-bio.
```

Comment:
This section of the documentation (*User* and *Developer*) is a little awkward. I think we should reorganize this a bit, but not sure if it can happen before beta.